nix/packages/grub-efi-image.nix

```diff
@@ -0,0 +1,51 @@
+{ lib, stdenvNoCC, grub2_efi, fetchFromGitHub, git, gnulib }:
+let
+  grub_uki = grub2_efi.overrideAttrs (finalAttrs: previousAttrs: rec {
+    patches = [ ];
+    src = fetchFromGitHub {
+      owner = "GovanifY";
+      repo = "grub";
+      rev = "master";
+      sha256 = "sha256-2XhZdRCK73QJ5QIeTdqXVFinzXq36pJ/6EwnluY5rnA=";
+    };
```

Comment:
What changes does your Grub fork have? Do you have an idea about how these could be upstreamed?

Reply:
The fork is currently not necessary per se, this was more of an experiment. The patch adds some of systemd's boot loader interface, but are really unneeded, I'll probably remove it. The more pressing concern is that GRUB does not implement SimpleFS which is used in the stub and thus causes it to panic. 

Correct me if I'm wrong too but it seems like EmbeddedConfig reads back the binary from the filesystem after it was verified by SB with StartImage? If so this is a TOCTOU in the making as anything that could MITM the filesystem could first return a correctly signed stub which then loads back its config from an edited version after that.